js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,98 @@
+describe("enhancedDate", function ()  {
+  'use strict';
+
+  describe("setDate", function () {
+
+    it("Should instance of date", function () {
+      expect(enhancedDate.setDate(new Date(2015, 4, 23))).to.be.a("Date");
+    });
+
+    it("Should return: " + (new Date(2015, 4, 23)), function () {
+      expect(enhancedDate.setDate(new Date(2015, 4, 23))).to.be.deep.equal(new Date(2015, 4, 23));
+    });
+  });
+
+  describe("getDate",function () {
+
+    it("Should be typeof number", function () {
+      enhancedDate.setDate(new Date(2015, 4, 23));
+      expect(enhancedDate.getDate()).to.be.a("number");
+    });
+
+    it("Should be equal to: " + (new Date(2015, 4, 23)), function () {
+      enhancedDate.setDate(new Date(2015, 4, 23));
+      expect(enhancedDate.getDate()).to.be.equal((new Date(2015, 4, 23)).getTime());
+    });
+  });
+
+  describe("getDayName",function () {
+
+    it("Should be typeof string", function () {
+      enhancedDate.setDate(new Date(2015, 4, 23));
+      expect(enhancedDate.getDayName()).to.be.a("string");
+    });
+
+    it("Should be equal to Wednesday", function () {
+      enhancedDate.setDate(new Date(2015, 4, 23));
+      expect(enhancedDate.getDayName()).to.be.equal("Saturday");
+    });
+
+    it("Should not be equal to Saturday", function () {
+      enhancedDate.setDate(new Date(2015, 4, 23));
+      expect(enhancedDate.getDayName()).to.not.be.equal("Monday");
```

Comment:
Once you've validated a string output like you have, the not assertion doesn't really add any value. It would probably be more useful to simply test other dates that resulted in other day names.